src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -369,6 +370,73 @@ public void SetModelValue([NotNull] string key, [NotNull] ValueProviderResult va
             GetModelStateForKey(key).Value = value;
         }
 
+        /// <summary>
+        /// Clears <see cref="ModelStateDictionary"/> entries for model in specified <see cref="ModelMetadata"/>.
+        /// </summary>
+        /// <param name="modelMetadata">The <see cref="ModelMetadata"/>.</param>
+        /// <param name="modelKey">The entry to clear. </param>
+        /// <param name="modelMetadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
+        public void ClearModelStateDictionaryForModel(
```

Comment:
A. Suspect we agree here.  My comment was specifically about "The onus of passing in a correct (possibly non empty key) should be on the caller."  Sorry that wasn't clear.
B. taking offline